### PR TITLE
Feature/#2056 style calendar meeting

### PR DESCRIPTION
**Pull request by aminbksi**

### CODE REVIEW

File: src/scenes/meeting/pages/MeetingRoomPage/components/templates/MeetingUser/components/UserMenu/UserMenu.tsx [hunk 1]
```diff
@@ -11,10 +11,11 @@ interface PropsInterface {
   onMuteUser?: () => void;
   onKickUser?: () => void;
   onClose?: () => void;
+  coords: {left: number; top: number; width: number};
```
Comment (PAKovalev) [inline src/scenes/meeting/pages/MeetingRoomPage/components/templates/MeetingUser/components/UserMenu/UserMenu.tsx:14]: Create interface

File: src/ui-kit/hooks/useUpdateCoordinates.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import {RefObject, useCallback, useState} from 'react';
+
+import {useScroll} from './useScroll';
+import {useResize} from './useResize';
+
+const OFFSET_RIGHT = 182;
```
Comment (PAKovalev) [inline src/ui-kit/hooks/useUpdateCoordinates.ts:6]: Make it as params. Move constants to components

File: src/ui-kit/hooks/useUpdateCoordinates.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+const OFFSET_BOTTOM = 7;
+
+const useUpdateCoordinates = (ref: RefObject<HTMLElement>) => {
+  const [coords, setCoords] = useState({left: 0, top: 0, width: 0});
```
Comment (PAKovalev) [inline src/ui-kit/hooks/useUpdateCoordinates.ts:10]: Use interface

File: src/ui-kit/hooks/useUpdateCoordinates.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  const [isMenuShown, setIsMenuShown] = useState<boolean>(false);
+
+  const updateCoordsOnScroll = () => {
```
Comment (PAKovalev) [inline src/ui-kit/hooks/useUpdateCoordinates.ts:13]: Don't need to create separate function.

File: src/ui-kit/hooks/useUpdateCoordinates.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    }
+  }, [ref]);
+
+  useScroll(ref, updateCoordsOnScroll);
+  useResize(ref, updateCoords);
+
+  return {coords, isMenuShown, updateCoords, setIsMenuShown};
+};
+
+export {useUpdateCoordinates};
```
Comment (PAKovalev) [inline src/ui-kit/hooks/useUpdateCoordinates.ts:30]: just `useCoordinates`

File: src/ui-kit/hooks/useUpdateCoordinates.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+import {useScroll} from './useScroll';
+import {useResize} from './useResize';
+
+const OFFSET_RIGHT = 182;
+const OFFSET_BOTTOM = 7;
+
+const useUpdateCoordinates = (ref: RefObject<HTMLElement>) => {
+  const [coords, setCoords] = useState({left: 0, top: 0, width: 0});
+  const [isMenuShown, setIsMenuShown] = useState<boolean>(false);
```
Comment (PAKovalev) [inline src/ui-kit/hooks/useUpdateCoordinates.ts:11]: isShown. Don't use Menu word. It is reusable hook.

**Review by PAKovalev — CHANGES_REQUESTED** [review]

**Review by PAKovalev — APPROVED** [review]

**Review by dmitry-yudakov — APPROVED** [review]

File: src/ui-kit/hooks/useCoordinates.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+const useCoordinates = (ref: RefObject<HTMLElement>, rightOffset: number, bottomOffset: number) => {
+  const [coords, setCoords] = useState<CoordinationInterface>();
+  const [isShown, setIsShown] = useState<boolean>(false);
+
+  const updateCoords = useCallback(() => {
+    const rect = ref?.current?.getBoundingClientRect();
+    if (rect) {
+      setCoords({left: rect.x - rightOffset, top: rect.y - bottomOffset, width: rect.width});
+    }
+  }, [ref]);
```
Comment (dmitry-yudakov) [inline src/ui-kit/hooks/useCoordinates.ts:17]: Sorry, it's a bit late on accepted PR, but we should probably add bottomOffset/rightOffset to deps list in this useCallback.

Comment (aminbksi) [reply src/ui-kit/hooks/useCoordinates.ts:17]: yes true, I added offsets first as constant in this hook but later made it as props, yeah I fixed it and will be merged in the next PR

**Pull request merged**